packages/myst-config/src/errorRules/validators.ts

```diff
@@ -0,0 +1,63 @@
+import type { ValidationOptions } from 'simple-validators';
+import {
+  defined,
+  incrementOptions,
+  validateChoice,
+  validateList,
+  validateObjectKeys,
+  validateString,
+} from 'simple-validators';
+import type { ErrorRule } from './types.js';
+
+const ERROR_RULE_KEY_OBJECT = {
+  required: ['id'],
+  optional: ['severity'],
+  alias: {
+    rule: 'id',
+  },
+};
+
+export function validateErrorRule(input: any, opts: ValidationOptions): ErrorRule | undefined {
+  if (typeof input === 'string') {
+    input = { id: input };
+  }
+  const value = validateObjectKeys(input, ERROR_RULE_KEY_OBJECT, {
+    ...opts,
+    suppressWarnings: true,
+    keepExtraKeys: true,
+  });
+  if (value === undefined) return undefined;
+  const id = validateString(value.id, incrementOptions('id', opts));
+  if (!id) return undefined;
+  const output: ErrorRule = { id };
+  if (defined(value.severity)) {
+    const choice = validateChoice(value.severity, {
+      ...incrementOptions('severity', opts),
+      choices: ['ignore', 'warn', 'error'],
```

Comment:
🤔 Would be nice if `validateChoice` had an option for `caseInsensitive`... (out of scope of this PR!)